lib/publiccloud/instance.pm

```diff
@@ -203,10 +203,60 @@ sub wait_for_guestregister
         }
         sleep 1;
     }
+
     $self->upload_log('/var/log/cloudregister', log_name => $autotest::current_test->{name} . '-cloudregister.log');
     die('guestregister didn\'t end in expected timeout=' . $args{timeout});
 }
 
+=head2 check_guestregister
+
+    check_guestregister();
+
+Check that the registration state of newly created guest match our requirements.
+https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/integration_test-process.txt
+=cut
+sub check_guestregister {
+    my ($self) = @_;
+
+    if (is_byos) {
+        if ($self->run_ssh_command(cmd => 'sudo zypper lr', proceed_on_failure => 1) !~ /No repositories defined/gm) {
+            die 'The BYOS instance should be unregistered and report "Warning: No repositories defined.".';
+        }
+
+        if ($self->run_ssh_command(cmd => 'sudo systemctl is-enabled guestregister.service', proceed_on_failure => 1) !~ /disabled/) {
+            die('guestregister.service is not disabled');
+        }
+
+        if ($self->run_ssh_command(cmd => 'sudo ls /etc/zypp/credentials.d/ | wc -l', proceed_on_failure => 1, quiet => 1) != 0) {
+            die('/etc/zypp/credentials.d/ is not empty');
+        }
+
+        if ($self->run_ssh_command(cmd => 'sudo stat --printf="%s" /var/log/cloudregister', proceed_on_failure => 1, quiet => 1) != 0) {
+            die('/var/log/cloudregister is not empty');
+        }
+    } else {
+        if ($self->run_ssh_command(cmd => 'sudo zypper lr | wc -l', timeout => 360) < 5) {
```

Comment:
Forgotten `quiet`? You can ignore this comment if this is only a matter of style without functional implications.

Reply:
TBH I'd put `quiet => 0` everywhere. That would make the review easier but also the code even less readable. WDYT?

Reply:
Yes, this is the only command where it is missing. I was wondering if that was on purpose of simply forgotten.